rust/stracciatella_c_api/src/c/librarydb.rs

```diff
@@ -32,19 +32,14 @@ pub extern "C" fn LibraryDB_push(
     data_dir: *const c_char,
     library: *const c_char,
 ) -> bool {
+    forget_rust_error();
     let ldb = unsafe_mut(ldb);
     let data_dir = path_from_c_str_or_panic(unsafe_c_str(data_dir));
     let library = path_from_c_str_or_panic(unsafe_c_str(library));
-    match ldb.add_library(data_dir, library) {
-        Err(err) => {
-            remember_rust_error(format!("{:?}", err));
-            false
-        }
-        Ok(_) => {
-            forget_rust_error();
-            true
-        }
+    if let Err(err) = ldb.add_library(data_dir, library) {
+        remember_rust_error(format!("{:?}", err));
```

Comment:
Might be good to include data_dir and library in the error message.

Reply:
Done